py/src/braintrust/util.py

```diff
@@ -50,3 +52,31 @@ def get_caller_location():
 # https://stackoverflow.com/questions/5574702/how-do-i-print-to-stderr-in-python.
 def eprint(*args, **kwargs):
     print(*args, file=sys.stderr, **kwargs)
+
+
+T = TypeVar("T")
+
+
+class LazyValue(Generic[T]):
```

Comment:
why not just use `cached_property` from functools?

https://docs.python.org/3/library/functools.html#functools.cached_property

Since our minimum python version is 3.8, we should have it everywhere

Reply:
Yeah there is some good stuff in functools. If I understand correctly, `@cached_property` seems specifically intended for "getter/setter" methods in classes (not really what we're doing). It's also not thread-safe, so we'd need to include that ourselves. Maybe `@cached`/`@lru_cache` is closer to what we want, but I feel like they are a bit overkill (we don't need arguments, and we don't always need thread safety).

I thought that this was a simple-enough util that I'd rather roll it myself.

Reply:
Yeah that makes sense / works for me